OSDOCS#7763: Updating OSD documentation to be cloud agnostic or have GCP references included

eohartman opened the pull request:
Issue:
[OSDOCS-7763](https://issues.redhat.com/browse/OSDOCS-7763)

Version:
4.15+

Link to docs preview:

1. [General settings table: Balancing-ignored-labels setting uses row](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_cluster_admin/osd-cluster-autoscaling.html#general-settings) - This no longer references AWS.
2. [Information collected by the Insights Operator](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/support/remote_health_monitoring/about-remote-health-monitoring#insights-operator-what-information-is-collected_about-remote-health-monitoring) - The fifth bullet point no longer mentions AWS. Additionally, removed a period from another bullet point so they're all consistent.
3. [Using bound service account tokens](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/authentication/bound-service-account-tokens) - Added GCP IAM to first paragraph.
4. [Example trigger authentication with pod authentication providers](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/nodes/cma/nodes-cma-autoscaling-custom-trigger-auth) - Added 'gcp-gke' value to 3 Specifies a pod identity.
5. [Log output types](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/log_collection_forwarding/logging-output-types#logging-output-types-descriptions) - added cloudlogging into the Output type descriptions section.
6. [Enabling stream-based retention for Loki](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/log_storage/cluster-logging-loki#logging-loki-retention_cluster-logging-loki) - Added two examples - Example global stream-based retention for GCP and Example per-tenant stream-based retention for GCP.
7. [Cloudwatch recommendation for OpenShift Dedicated](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/cluster-logging#cluster-logging-about-custom-resources_cluster-logging) - Removed "Cloudwatch recommendation for OpenShift Dedicated section.


The following have a note added above the example:

1. [oc adm release extract](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/cli_reference/openshift_cli/administrator-cli-commands#oc-adm-release-extract) - Added note before the example.
2. [oc image mirror](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/cli_reference/openshift_cli/developer-cli-commands#oc-image-mirror) - Added note before the example.
3. [Showing data collected by Telemetry](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/support/remote_health_monitoring/showing-data-collected-by-remote-health-monitoring#showing-data-collected-from-the-cluster_showing-data-collected-by-remote-health-monitoring) - Added note before the example.
4. [Understanding how to use pod disruption budgets to specify the number of pods that must be up](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/nodes/pods/nodes-pods-configuring#nodes-pods-pod-distruption-about_nodes-pods-configuring) - Added note before the example.

Note: Before and after images are attached to [OSDOCS-7763](https://issues.redhat.com/browse/OSDOCS-7763) for ease of viewing.

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

eohartman commented:
![Example per-tenant stream-based retention for GCP](https://github.com/openshift/openshift-docs/assets/107144651/576209b7-ffde-4cbf-84d9-bb2660d858f9)
![Example global stream-based retention for GCP](https://github.com/openshift/openshift-docs/assets/107144651/ae19ab47-3b62-47a9-89b6-24e0385ac5c4)
![CloudWatch recommendation before](https://github.com/openshift/openshift-docs/assets/107144651/356684f8-df00-4c9a-bca7-4c4f9bb3469b)
![CloudWatch recommendation after](https://github.com/openshift/openshift-docs/assets/107144651/afc5fdbd-a377-4239-96a4-a33fd28c2db2)
![output types description after](https://github.com/openshift/openshift-docs/assets/107144651/b423ec58-7cf4-413b-879e-6e0e44e7b4d3)
![output types description before](https://github.com/openshift/openshift-docs/assets/107144651/06e0d3b8-6455-4226-8c9d-05970ec5a027)
![example trigger authentication after](https://github.com/openshift/openshift-docs/assets/107144651/a2d11fb4-5ff7-4419-9f48-26ba127c0b14)
![example trigger authentication before](https://github.com/openshift/openshift-docs/assets/107144651/168fa812-8bbe-4582-8946-474a10ef605b)
![pod disruption after](https://github.com/openshift/openshift-docs/assets/107144651/51334d7c-0020-4ef5-9bf9-bd9449b24292)
![pod disruption before](https://github.com/openshift/openshift-docs/assets/107144651/cfa10821-fa2d-44e4-9d03-f45709523d5a)
![Showing data collected by Telemetry before](https://github.com/openshift/openshift-docs/assets/107144651/e732c4df-f25b-4f02-806a-6a9e724aef88)
![ocm adm release extract before](https://github.com/openshift/openshift-docs/assets/107144651/8916191a-3fc0-4251-aa26-d52338132c03)
![oc image mirror after](https://github.com/openshift/openshift-docs/assets/107144651/6f81e69e-a12d-4b63-a8e0-0dba83392624)
![oc image mirror before](https://github.com/openshift/openshift-docs/assets/107144651/5b66818a-2407-4fda-9d97-01339e3d7e0e)
![Scope of the network verification checks after](https://github.com/openshift/openshift-docs/assets/107144651/912fc766-9220-4e73-808c-d441a0fffcb5)
![Scope of the network verification checks before](https://github.com/openshift/openshift-docs/assets/107144651/31e818bf-f304-4d08-af18-970089d68566)
![Using bound service account tokens after](https://github.com/openshift/openshift-docs/assets/107144651/78eccc3d-4a9d-4369-8e7c-56342225b9f0)
![Using bound service account tokens before](https://github.com/openshift/openshift-docs/assets/107144651/a00e6804-dd36-4128-aff0-1ace0555f889)
![Information collected by the Insights Operator after](https://github.com/openshift/openshift-docs/assets/107144651/b44d6b63-a089-4c13-a0b4-809e7e972154)
![Information collected by the Insights Operator before](https://github.com/openshift/openshift-docs/assets/107144651/c87c40f0-7839-4c38-bc6e-d2f0c06bbc80)
![balancing-ignored-labels-before](https://github.com/openshift/openshift-docs/assets/107144651/fb0be397-4233-4dd3-ab26-32a767477e59)
![balancing-ignored-labels-after](https://github.com/openshift/openshift-docs/assets/107144651/10f0bc38-c7d6-4cbd-8596-b8741c3ee8a0)
![ocm adm release extract after](https://github.com/openshift/openshift-docs/assets/107144651/7b2ae6fe-440b-4262-af6d-669b29d958fc)
![Showing data collected by Telemetry after](https://github.com/openshift/openshift-docs/assets/107144651/e145cf94-0980-4fe3-a2a4-1cd2d313a3ca)

eohartman commented:
@lento-sun @shreyansvm @abhgupta Can y'all please take a look at this PR? I still have a couple more questions that need answering, but I'd like to go ahead and at least get a preliminary review. 

Thank you!

eohartman commented:
@yuwang-RH  Can you please QE this PR? Thank you!

svmrh commented:
Network Verifier feature is today supported on ROSA and OSD-AWS, but NOT on OSD-GCP. There is a roadmap item to support this during 2H-2025. Here's the feature tracking Network Verifier support on OSD-GCP - XCMSTRAT-477
> - [Scope of the network verification checks](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/networking/network-verification#scope-of-the-network-verification-checks_network-verification) - The last bullet point now includes OCP reference and link, as well.

As per OCP 4.15 documentation, "gcp-gke" is not a supported value for PodIdentity. Reference [OCP 4.15 documentation](https://docs.openshift.com/container-platform/4.15/nodes/cma/nodes-cma-autoscaling-custom-trigger-auth.html). 
> - [Example trigger authentication with pod authentication providers](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/nodes/cma/nodes-cma-autoscaling-custom-trigger-auth) - Added 'gcp-gke' value to 3 Specifies a pod identity.

This link is broken. So couldn't review. 
> - [Log output types](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/logging/log_collection_forwarding/logging-output-types#logging-output-types-descriptions) - added cloudlogging into the Output type descriptions section.
<img width="1098" alt="image" src="https://github.com/openshift/openshift-docs/assets/130616558/5bf88b22-3c94-4519-9405-3b5f2fb2634e">


These updates look okay

> - [General settings table: Balancing-ignored-labels setting uses row](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_cluster_admin/osd-cluster-autoscaling.html#general-settings) - This no longer references AWS.
> - [Using bound service account tokens](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/authentication/bound-service-account-tokens) - Added GCP IAM to first paragraph.

@lento-sun, @xueli181114  / QE team : can you please confirm the following along with rest of the review?

> - [Information collected by the Insights Operator](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/support/remote_health_monitoring/about-remote-health-monitoring#insights-operator-what-information-is-collected_about-remote-health-monitoring) - The fifth bullet point no longer mentions AWS. Additionally, removed a period from another bullet point so they're all consistent.
> - [Enabling stream-based retention for Loki](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/log_storage/cluster-logging-loki#logging-loki-retention_cluster-logging-loki) - Added two examples - Example global stream-based retention for GCP and Example per-tenant stream-based retention for GCP.

"CloudWatch recommendation for OpenShift Dedicated" was a section header. Now that this is removed, QE team - can you please check if the rest of the flow, logging requirements and configurations look okay? 
> - [Cloudwatch recommendation for OpenShift Dedicated](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/cluster-logging#cluster-logging-about-custom-resources_cluster-logging) - Removed "Cloudwatch recommendation for OpenShift Dedicated section.

These look okay. QE team, do you have corresponding GCP examples that we can add here?
> - [oc adm release extract](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/cli_reference/openshift_cli/administrator-cli-commands#oc-adm-release-extract) - Added note before the example.
> - [oc image mirror](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/cli_reference/openshift_cli/developer-cli-commands#oc-image-mirror) - Added note before the example.
> - [Showing data collected by Telemetry](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/support/remote_health_monitoring/showing-data-collected-by-remote-health-monitoring#showing-data-collected-from-the-cluster_showing-data-collected-by-remote-health-monitoring) - Added note before the example.
> - [Understanding how to use pod disruption budgets to specify the number of pods that must be up](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/nodes/pods/nodes-pods-configuring#nodes-pods-pod-distruption-about_nodes-pods-configuring) - Added note before the example.

xueli181114 commented on nodes/cma/nodes-cma-autoscaling-custom-trigger-auth.adoc:
```diff
@@ -117,7 +117,7 @@ spec:
 <1> Specifies the namespace of the object you want to scale.
 <2> Specifies that this trigger authentication uses a platform-native pod authentication method for authorization.
-<3> Specifies a pod identity. Supported values are `none`, `azure`, `aws-eks`, or `aws-kiam`. The default is `none`.
+<3> Specifies a pod identity. Supported values are `none`, `azure`, `gcp-gke`, `aws-eks`, or `aws-kiam`. The default is `none`.
```
@xingxingxia Can you please help to confirm with this line?

eohartman replied:
@xueli181114 @xingxingxia Is this line incorrect for OSD, or has it not just been added to OCP yet? I understand OCP documentation doesn't have it, but I'd like to have clarification on whether gcp-gke is not actually available, vs we have a gap in OCP documentation that also needs to be updated.

jianli-wei replied:
@xueli181114 Sorry I'm not sure about it. Anyway, I took a look at OpenShift documentation [Understanding custom metrics autoscaler trigger authentications](https://docs.openshift.com/container-platform/4.15/nodes/cma/nodes-cma-autoscaling-custom-trigger-auth.html), in the section "_Example trigger authentication with pod authentication providers_", it's said `Supported values are none, azure, aws-eks, or aws-kiam`. I guess you could confirm with Nodes QE.

lyman9966 replied:
@sunilcio @asahay19 Can you help to check this?

joelsmith replied:
For the current version of Custom Metrics Autoscaler (2.12.1) these are the valid values:
```
none | azure | azure-workload | aws-eks | aws-kiam | gcp
```
(From https://keda.sh/docs/2.12/concepts/authentication/#re-use-credentials-and-delegate-auth-with-triggerauthentication)

eohartman replied:
@joelsmith Thank you so much!

xueli181114 commented on networking/network-verification.adoc:
```diff
@@ -38,7 +38,7 @@ The network verification includes checks for each of the following requirements:
 * The VPC has `enableDnsHostnames` enabled.
 ifdef::openshift-dedicated[]
-* Egress is available to the required domain and port combinations that are specified in the xref:../osd_planning/aws-ccs.adoc#osd-aws-privatelink-firewall-prerequisites_aws-ccs[AWS firewall prerequisites] section.
+* Egress is available to the required domain and port combinations for either {product-title} (ROSA) or Google Cloud Platform. For ROSA the required domain and port combinations are specified in the xref:../osd_planning/aws-ccs.adoc#osd-aws-privatelink-firewall-prerequisites_aws-ccs[AWS firewall prerequisites] section. For GCP, they are specified in the xref:../osd_planning/gcp-ccs.adoc#ccs-gcp-c ...
```
@lihongan Can you please help to confirm with this part?

lihongan replied:
@huiran0826 could you please help take a look at this? Thanks

huiran0826 replied:
It seems to me not ovn egress feature, but cloud level?  Not sure.   @zhaozhanqi @anuragthehatter  any ideas?

zhaozhanqi replied:
+1, it should be cloud level network

lihongan replied:
@xueli181114 they are cloud provider network related, I'm not sure if @yunjiang29 and @jianli-wei could help review ?

eohartman replied:
For what it's worth, the [Customer cloud procedures on GCP](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_planning/gcp-ccs#ccs-gcp-customer-procedure_gcp-ccs) topic was written and approved in a previous Jira ticket. Please let me know if I need to separate the bullet points between AWS and GCP.

kabirbhartiRH commented on modules/logging-create-loki-cr-cli.adoc:
```diff
@@ -54,7 +54,7 @@ It is not possible to change the number `1x` for the deployment size.
 // end::pre-5.9[]
 
```
@xueli181114 If we have a section on Loki Operator deployment for OSD, I suggest checking the below docs on Openshift since there are changes to how Loki will be deployed on STS clusters from Logging 5.9 onwards
https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#installing-log-storage-loki-sts

xueli181114 replied:
@kabirbhartiRH Do you mind to provide more info? Do you mean the doc content you pasted should be introduced?

kabirbhartiRH replied:
@xueli181114 On our OCM profiles job, I do not see an OSD template which has STS turned ON. Could you confirm if STS is available on OSD clusters running on AWS/Azure?

xueli181114 replied:
@kabirbhartiRH STS only works for AWS cluster. Yes we can support STS for OSD cluster, but STS mainly used by rosa.

kabirbhartiRH replied:
@xueli181114 In case we support STS on OSD, I think it's better to include deploying Loki in STS mode.

Deploying the Operator in STS/Workload Identity Federation mode on AWS and Azure: https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#installing-log-storage-loki-sts

Section to create Loki Storage secret needed for LokiStack:

AWS: https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#AWS_storage_STS_installing-log-storage

Azure: https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#Azure_storage_STS_installing-log-storage

xueli181114 replied:
Thank you @kabirbhartiRH . STS is supported by OSD but it is not showing in the official [doc](https://docs.openshift.com/dedicated/osd_install_access_delete_cluster/creating-an-aws-cluster.html). I would suggest another jira card to track STS for OSD support since it's a big change. And customers cannot create an sts OSD via UI.

kabirbhartiRH replied:
@eohartman Please note above comment :)

eohartman replied:
@kabirbhartiRH Got it! I'll work on seeing if we already have a Jira ticket related to this, and if not, create one. Thank you!

eohartman replied:
Marking as resolved, as this documentation gap will be covered under [OSDOCS-10494](https://issues.redhat.com/browse/OSDOCS-10494)

kabirbhartiRH replied:
@eohartman It seems this [section](https://docs.openshift.com/dedicated/observability/logging/log_storage/installing-log-storage.html#logging-identity-federation_installing-log-storage) is already present under OSD docs?

eohartman commented:
> Network Verifier feature is today supported on ROSA and OSD-AWS, but NOT on OSD-GCP. There is a roadmap item to support this during 2H-2025. Here's the feature tracking Network Verifier support on OSD-GCP - [XCMSTRAT-477](https://issues.redhat.com//browse/XCMSTRAT-477)
 
1. I have removed network verifier reference.

> > * [Example trigger authentication with pod authentication providers](https://75001--ocpdocs-pr.netlify.app/openshift-dedicated/latest/nodes/cma/nodes-cma-autoscaling-custom-trigger-auth) - Added 'gcp-gke' value to 3 Specifies a pod identity.

2. Apologies - this is what happens when multiple writers are working on projects simultaneously. I have corrected the link in the origin description.

eohartman commented:
@AndrewJones-RH Can you please help give advice / coordinate which QE contacts can help me with the outstanding questions? There are so many tagged in this PR, the related Jira ticket, and Google Docs that I'm not sure who the best contact is for source information.

Thank you!

mletalie reviewed (commented):
/label peer-review-in-progress

mletalie reviewed (commented):
/label peer-review-done
/lgtm
/remove-label peer-review-in-progress

eohartman commented on networking/network-verification.adoc:
```diff
@@ -38,7 +38,7 @@ The network verification includes checks for each of the following requirements:
 * The VPC has `enableDnsHostnames` enabled.
 ifdef::openshift-dedicated[]
-* Egress is available to the required domain and port combinations that are specified in the xref:../osd_planning/aws-ccs.adoc#osd-aws-privatelink-firewall-prerequisites_aws-ccs[AWS firewall prerequisites] section.
+* Egress is available to the required domain and port combinations for {product-title} (ROSA). For ROSA the required domain and port combinations are specified in the xref:../osd_planning/aws-ccs.adoc#osd-aws-privatelink-firewall-prerequisites_aws-ccs[AWS firewall prerequisites] section.
```
We should mention ROSA here, but not GCP yet because it's still in development. (But it will be listed here eventually too, I believe.)

eohartman commented on authentication/bound-service-account-tokens.adoc:
```diff
@@ -6,7 +6,7 @@ include::_attributes/common-attributes.adoc[]
 toc::[]
 
-You can use bound service account tokens, which improves the ability to integrate with cloud provider identity access management (IAM) services, such as AWS IAM.
+You can use bound service account tokens, which improves the ability to integrate with cloud provider identity access management (IAM) services, such as AWS IAM or GCP IAM.
```
Good catch!

eohartman commented on networking/cidr-range-definitions.adoc:
```diff
@@ -17,7 +17,7 @@ Machine CIDR ranges cannot be changed after creating your cluster.
 
 ifdef::openshift-rosa,openshift-dedicated[]
-When specifying subnet CIDR ranges, ensure that the subnet CIDR range is within the defined Machine CIDR. You must verify that the subnet CIDR ranges allow for enough IP addresses for all intended workloads, including at least eight IP addresses for possible AWS Load Balancers.
+When specifying subnet CIDR ranges, ensure that the subnet CIDR range is within the defined Machine CIDR. You must verify that the subnet CIDR ranges allow for enough IP addresses for all intended workloads for possible AWS Load Balancers or Google Load Balancers.
```
Good point. I actually removed the reference to both AWS and GCP because when looking at it again, since this is an ifdef for both OSD and ROSA, it makes sense to have it be more generic rather than have GCP mentioned in the ROSA documentation.

svmrh commented on modules/cluster-logging-cloudwatch.adoc:
```diff
@@ -6,10 +6,6 @@
-[id="cluster-logging-cloudwatch_{context}"]
-= CloudWatch recommendation for {product-title}
-
-Red Hat recommends that you use the AWS CloudWatch solution for your logging needs.
```
Yes, let's completely remove this sentence as this is too specific to AWS. 

Instead, can we point users to [Log collection and forwarding](https://docs.openshift.com/dedicated/observability/logging/log_collection_forwarding/log-forwarding.html) section to learn more about "Log Output Types", "Configuring Log Forwarding" and other topics related to this?

[1] [https://docs.openshift.com/dedicated/observability/logging/log_collection_forwarding/logging-output-types.html](https://www.google.com/url?q=https://docs.openshift.com/dedicated/observability/logging/log_collection_forwarding/logging-output-types.html&sa=D&source=docs&ust=1715298068385815&usg=AOvVaw3xpxMVNbm6XxPltCcNmmS9)

[2] [https://docs.openshift.com/dedicated/observability/logging/log_collection_forwarding/configuring-log-forwarding.html](https://www.google.com/url?q=https://docs.openshift.com/dedicated/observability/logging/log_collection_forwarding/configuring-log-forwarding.html&sa=D&source=docs&ust=1715298068385923&usg=AOvVaw1wcBLXVuU6BPBDKX139rsN)

eohartman replied:
I'll add a "For information, see..." link to Log collection and forwarding.

anpingli commented on modules/logging-loki-retention.adoc:
```diff
@@ -10,17 +10,15 @@ With Logging version 5.6 and higher, you can configure retention policies based
-spec:
-  limits:
-    global: <1>
+   global: <1>
```
It is better to use the default class name is gp2 in AWS.   storageClassName: gp2

anpingli replied:
retention is cloud irrelevant. I think one example is enough.

eohartman replied:
I have removed the GCP examples and changed storageClassName for AWS examples.

anpingli commented on modules/logging-create-loki-cr-cli.adoc:
```diff
@@ -54,7 +54,7 @@ It is not possible to change the number `1x` for the deployment size.
 
 // tag::5.9[]
-+
+
```
Shall we provide an example for GCP here?

anpingli replied:
Can we provide a step about how to create secrets?

anpingli replied:
https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#logging-loki-storage-aws_installing-log-storage

https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/installing-log-storage.html#logging-loki-storage-gcp_installing-log-storage

kabirbhartiRH replied:
@anpingli It is already present under the official OSD docs
https://docs.openshift.com/dedicated/observability/logging/log_storage/installing-log-storage.html#logging-loki-storage_installing-log-storage

eohartman replied:
@anpingli What would the example for GCP look like? I need more information if you want me to provide an example of it in here.

kabirbhartiRH replied:
@eohartman There is already an example on GCP under OSD docs. 
https://docs.openshift.com/dedicated/observability/logging/log_storage/installing-log-storage.html#logging-loki-storage-gcp_installing-log-storage

eohartman replied:
Yes. So does this mean no additional example is needed, and I can resolve this PR comment? 

I'm getting conflicting information because of all the QEs involved on this project. Just trying to do my best to be thorough. 😊 Thank you!

anpingli replied:
Yes.

anpingli commented on modules/logging-loki-retention.adoc:
```diff
@@ -46,9 +44,47 @@ spec:
+  storage:
+    schemas:
+    - effectiveDate: "2020-10-11"
+      version: v11
```
It is better to use same version in all Lokistack examples and explain what the version is?  For 5.8 and prior, we can use v11,v12. from 5.9: we can use v13.   @kabirbhartiRH  correct me if I am wrong?

kabirbhartiRH replied:
Yes, for Logging 5.9 onwards schemas version should be v13. Prior to that, v11 (default) and v12 can be chosen.

eohartman replied:
Both AWS Lokistack examples reference v11. 

Should there be a note that explains that logging for 5.8 and prior, v11 or v12 should be the designated version number, but from 5.9+, it should be v13?

kabirbhartiRH replied:
5.9+ still supports v12 but v13 is the recommended schema version for 5.9+
It would be a good to have note.

eohartman replied:
Added "Although logging version 5.9 and higher supports schema v12, v13 is recommended." as a note.

kabirbhartiRH commented on modules/logging-loki-retention.adoc:
```diff
@@ -46,9 +44,47 @@ spec:
+metadata:
+  name: logging-loki
+  namespace: openshift-logging
+   global: <1>
```
This should be under the `spec` property. `spec` is missing here
For ref. https://docs.openshift.com/container-platform/4.15/observability/logging/log_storage/cluster-logging-loki.html#logging-loki-retention_cluster-logging-loki

kabirbhartiRH replied:
Also @eohartman, quick question regarding the point 6 on your PR description, We already have a section on retention under official [docs](https://docs.openshift.com/dedicated/observability/logging/log_storage/cluster-logging-loki.html#logging-loki-retention_cluster-logging-loki) Any reason why we are adding it again for GCP? It seems redundant.

eohartman replied:
Added the `spec` property and removed GCP examples.

eohartman commented:
/retest

anpingli commented:
/lgtm

EricPonvelle commented:
/cherrypick enterprise-4.15

EricPonvelle commented:
/cherrypick enterprise-4.16
